NPE in ClassTransformingBuildStep when building multi-module application

**Describe the bug**
I have a multi-module application, set up like this:

- `core` is almost the entire application, except of a JDBC driver, and it doesn't have the `quarkus-maven-plugin`
    - it uses the `quarkus-hibernate-orm-panache` and `quarkus-openshift` extensions
- a `mariadb` module adds the MariaDB JDBC driver and runs the `quarkus-maven-plugin`
- there's also a bunch of other modules for other databases, but those are not important here

When I try to build the app on latest Quarkus `master`, I get this NPE during the `quarkus-maven-plugin` execution:

```
[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:999-SNAPSHOT:build (default) on project sql-db-mariadb: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
[ERROR]         [error]: Build step io.quarkus.deployment.steps.ClassTransformingBuildStep#handleClassTransformation threw an exception: java.lang.IllegalStateException: java.util.concurrent.ExecutionException: java.lang.NullPointerException
[ERROR]         at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:947)
[ERROR]         at io.quarkus.builder.BuildContext.run(BuildContext.java:273)
[ERROR]         at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2027)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1551)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1442)
[ERROR]         at java.lang.Thread.run(Thread.java:748)
[ERROR]         at org.jboss.threads.JBossThread.run(JBossThread.java:479)
[ERROR] Caused by: java.util.concurrent.ExecutionException: java.lang.NullPointerException
[ERROR]         at java.util.concurrent.FutureTask.report(FutureTask.java:122)
[ERROR]         at java.util.concurrent.FutureTask.get(FutureTask.java:192)
[ERROR]         at io.quarkus.deployment.steps.ClassTransformingBuildStep.handleClassTransformation(ClassTransformingBuildStep.java:103)
[ERROR]         at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR]         at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[ERROR]         at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR]         at java.lang.reflect.Method.invoke(Method.java:498)
[ERROR]         at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:938)
[ERROR]         ... 7 more
[ERROR] Caused by: java.lang.NullPointerException
[ERROR]         at java.nio.file.Files.provider(Files.java:97)
[ERROR]         at java.nio.file.Files.readAttributes(Files.java:1737)
[ERROR]         at java.nio.file.Files.size(Files.java:2332)
[ERROR]         at io.quarkus.deployment.steps.ClassTransformingBuildStep$1.call(ClassTransformingBuildStep.java:73)
[ERROR]         at io.quarkus.deployment.steps.ClassTransformingBuildStep$1.call(ClassTransformingBuildStep.java:67)
[ERROR]         at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[ERROR]         at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[ERROR]         at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[ERROR]         at java.lang.Thread.run(Thread.java:748)
```

This used to work just fine fairly recently, so I found a working commit and did a `git bisect` to find out the problem. Bisecting led me to commit `403bf6c960eb9e4fc2fb84a93f735dd95f950c21`. I have then turned back to `master`, reverted the commit and confirmed that the application builds just fine again.

I have also added a couple of breakpoints to `ClassTransformingBuildStep` and found that, in my case, the `appArchives.containingArchive` call returns a `dekorate-dependencies` archive for some `io.fabric8.kubernetes.client` classes. The `dekorate-dependencies` JAR does _not_ include Fabric8 Kubernetes Client classes, yet for some reason, the index for that JAR does include them.

**Expected behavior**
Multi-module build works OK.

**Actual behavior**
Multi-module build fails with a NPE.

**To Reproduce**
I have extracted a minimal reproducer into this repo: https://github.com/Ladicek/quarkus-multimodule-build-npe

Steps to reproduce the behavior:
1. `git clone https://github.com/Ladicek/quarkus-multimodule-build-npe.git`
2. `mvn clean package`
3. Observe NPE.

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: `Linux argondie 4.15.0-91-generic #92-Ubuntu SMP Fri Feb 28 11:09:48 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux`
 - Output of `java -version`:
```
openjdk version "1.8.0_242"
OpenJDK Runtime Environment (build 1.8.0_242-8u242-b08-0ubuntu3~18.04-b08)
OpenJDK 64-Bit Server VM (build 25.242-b08, mixed mode)
```
 - GraalVM version (if different from Java): N/A
 - Quarkus version or git rev: `master`, specifically `863e9a6a73d9c60f5a92665e69e80da3adfa9f94`
 - Build tool (ie. output of `mvnw --version` or `gradlew --version`):
```
Apache Maven 3.6.3 (cecedd343002696d0abb50b32b541b8a6ba2883f)
Maven home: /home/lthon/software/apache-maven
Java version: 1.8.0_242, vendor: Private Build, runtime: /usr/lib/jvm/java-8-openjdk-amd64/jre
Default locale: cs_CZ, platform encoding: UTF-8
OS name: "linux", version: "4.15.0-91-generic", arch: "amd64", family: "unix"
```

--- a/bom/runtime/pom.xml
+++ b/bom/runtime/pom.xml
@@ -143,7 +143,7 @@
         <awssdk.version>2.10.70</awssdk.version>
         <azure-functions-java-library.version>1.3.0</azure-functions-java-library.version>
         <kotlin.version>1.3.71</kotlin.version>
-        <dekorate.version>0.11.4</dekorate.version>
+        <dekorate.version>0.11.5</dekorate.version>
         <maven-artifact-transfer.version>0.10.0</maven-artifact-transfer.version>
         <jline.version>2.14.6</jline.version>
         <maven-invoker.version>3.0.1</maven-invoker.version>

